### PR TITLE
Driver reshuffle

### PR DESCRIPTION
**Pull request by KurochkaR**
test after pr #387

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -392,7 +392,7 @@ class Schema(models.TextChoices):
         BUYER = 'BUYER', 'Схема під викуп'
         CUSTOM = 'CUSTOM', 'Індивідуальна схема'
 
-    fleet = models.OneToOneField('Fleet', blank=True, null=True, on_delete=models.SET_NULL, verbose_name='Автопарк')
+    # fleet = models.OneToOneField('Fleet', blank=True, null=True, on_delete=models.SET_NULL, verbose_name='Автопарк')
```
Comment (SpaceIgor) [inline app/models.py:395]: why?

File: auto_bot/handlers/order/handlers.py [hunk 2]
```diff
@@ -168,11 +168,10 @@ def from_address(update, context):
 
 
 def to_the_address(update, context):
-
     query = update.callback_query
     chat_id = update.effective_chat.id
-    state = int(redis_instance().hget(str(chat_id), 'state'))
-    if state == FROM_ADDRESS:
+    state = redis_instance().hget(str(chat_id), 'state')
+    if state and state == str(FROM_ADDRESS):
```
Comment (SpaceIgor) [inline auto_bot/handlers/order/handlers.py:174]: back to int

File: auto_bot/handlers/order/handlers.py [hunk 3]
```diff
@@ -201,8 +200,8 @@ def to_the_address(update, context):
 def payment_method(update, context):
     query = update.callback_query
     chat_id = update.effective_chat.id
-    state = int(redis_instance().hget(str(chat_id), 'state'))
-    if state == TO_THE_ADDRESS:
+    state = redis_instance().hget(str(chat_id), 'state')
+    if state and state == str(TO_THE_ADDRESS):
```
Comment (SpaceIgor) [inline auto_bot/handlers/order/handlers.py:204]: back to int

File: selenium_ninja/synchronizer.py [hunk 4]
```diff
@@ -22,138 +24,106 @@ def synchronize(self):
...
             vehicle.vin_code = vin_code
-            update_fields.append('vin_code')
-        if update_fields:
-            vehicle.save(update_fields=update_fields)
 
-    @staticmethod
-    def update_driver_fields(driver, **kwargs):
-        key_phone, key_email = 'phone_number', 'email'
-        update_fields, phone_number, email = [], kwargs[key_phone], kwargs[key_email]
+        if vehicle_name or vin_code:
```
Comment (SpaceIgor) [inline selenium_ninja/synchronizer.py:108]: any()

File: selenium_ninja/synchronizer.py [hunk 5]
```diff
@@ -22,138 +24,106 @@ def synchronize(self):
...
-            update_fields.append(key_phone)
-        if driver.email != email and email:
+
+        if email and driver.email != email:
             driver.email = email
-            update_fields.append(key_email)
-        if update_fields:
-            driver.save(update_fields=update_fields)
+
+        if phone_number or email or vehicle:
```
Comment (SpaceIgor) [inline selenium_ninja/synchronizer.py:125]: any()

File: auto/tasks.py [hunk 6]
```diff
@@ -43,6 +44,31 @@
 
 logger = get_task_logger(__name__)
 
+fleets = {
```
Comment (SpaceIgor) [inline auto/tasks.py:47]: need models with options for partner and relation on fleet
fleet add when - has been auth and we can start tasks on fleet

**Review by OlegBondarUA — APPROVED** [review]

File: auto_bot/handlers/order/handlers.py [hunk 7]
```diff
@@ -253,9 +253,11 @@ def get_additional_info(update, context):
...
     query = update.callback_query
     chat_id = update.effective_chat.id
-    data = int(query.data.split(' ')[1])
-    response = query.message.reply_markup.inline_keyboard[data][0].text
-    if data:
+    callback = query.data
+    if callback == "Return_info":
+        payment_method(update, context)
+    elif int(callback.split(' ')[1]):
+        response = query.message.reply_markup.inline_keyboard[int(callback.split(' ')[1])][0].text
```
Comment (SpaceIgor) [inline auto_bot/handlers/order/handlers.py:260]: u can use simple split()

File: auto_bot/handlers/order/keyboards.py [hunk 8]
```diff
@@ -66,7 +66,7 @@ def inline_payment_kb():
     keyboard = [
         [InlineKeyboardButton(price_inline_buttons[4], callback_data="Cash_payment 0")],
         [InlineKeyboardButton(price_inline_buttons[5], callback_data="Card_payment 1")],
-        [InlineKeyboardButton(order_inline_buttons[6], callback_data="Right_place")]
+        [InlineKeyboardButton(order_inline_buttons[6], callback_data="Return_info")]
```
Comment (SpaceIgor) [inline auto_bot/handlers/order/keyboards.py:69]: not correct callback

File: auto_bot/handlers/order/static_text.py [hunk 9]
```diff
@@ -173,10 +173,12 @@ def personal_order_info(order):
     return message
 
 
-def client_order_info(order):
+def client_order_info(order, time_update=None):
     if order.car_delivery_price:
         message = f"Замовлення оновлено\n" \
                   f"Нова сума замовлення: {order.sum} грн\n"
+    elif time_update:
```
Comment (SpaceIgor) [inline auto_bot/handlers/order/static_text.py:180]: retry use var in start func and put in logic

File: selenium_ninja/synchronizer.py [hunk 10]
```diff
@@ -22,138 +24,106 @@ def synchronize(self):
...
-        try:
-            driver = self.get_driver_by_name(kwargs['name'],
-                                             kwargs['second_name'],
-                                             partner=self.partner_id)
-        except ObjectDoesNotExist:
+        driver = Driver.objects.filter((Q(name=kwargs['name'], second_name=kwargs['second_name']) |
+                                        Q(name=kwargs['second_name'], second_name=kwargs['name']) |
+                                        Q(phone_number__icontains=kwargs['phone_number'][-10:])) &
+                                       Q(partner=self.partner_id)).first()
+        if not driver:
```
Comment (SpaceIgor) [inline selenium_ninja/synchronizer.py:58]: not correct create driver - without partner

File: app/models.py (hunk 1 above)
Comment (KurochkaR) [reply app/models.py:395]: for what this field driver can use only one fleet and fleet can use only one driver???

File: auto_bot/handlers/order/handlers.py (hunk 2 above)
Comment (KurochkaR) [reply auto_bot/handlers/order/handlers.py:174]: not working int(NONE)

File: auto_bot/handlers/order/handlers.py (hunk 3 above)
Comment (KurochkaR) [reply auto_bot/handlers/order/handlers.py:204]: same

File: selenium_ninja/synchronizer.py (hunk 4 above)
Comment (KurochkaR) [reply selenium_ninja/synchronizer.py:108]: for 2 variables?

File: selenium_ninja/synchronizer.py (hunk 5 above)
Comment (KurochkaR) [reply selenium_ninja/synchronizer.py:125]: here ok

File: auto_bot/handlers/order/keyboards.py (hunk 8 above)
Comment (KurochkaR) [reply auto_bot/handlers/order/keyboards.py:69]: it's my callback how it can be not correct??oO

Comment (KurochkaR) [reply auto_bot/handlers/order/keyboards.py:69]: in payment i must return to additional info

File: selenium_ninja/synchronizer.py (hunk 10 above)
Comment (KurochkaR) [reply selenium_ninja/synchronizer.py:58]: it's check driver in database not creating creating in if not driver statement

File: selenium_ninja/synchronizer.py (hunk 4 above)
Comment (SpaceIgor) [reply selenium_ninja/synchronizer.py:108]: +

File: auto_bot/handlers/order/handlers.py (hunk 2 above)
Comment (SpaceIgor) [reply auto_bot/handlers/order/handlers.py:174]: same

File: auto/tasks.py (hunk 6 above)
Comment (KurochkaR) [reply auto/tasks.py:47]: we start tasks if partner have passwords in parksettings for this fleet

File: auto_bot/handlers/order/keyboards.py (hunk 8 above)
Comment (SpaceIgor) [reply auto_bot/handlers/order/keyboards.py:69]: when u have info and push back step u will have error

Comment (KurochkaR) [reply auto_bot/handlers/order/keyboards.py:69]: qa test this no errors

File: selenium_ninja/synchronizer.py (hunk 4 above)
Comment (KurochkaR) [reply selenium_ninja/synchronizer.py:108]: less using memory using simple or instead any

File: auto_bot/handlers/order/static_text.py (hunk 9 above)
Comment (KurochkaR) [reply auto_bot/handlers/order/static_text.py:180]: it's needed for update time in time order

File: auto_bot/handlers/order/handlers.py (hunk 7 above)
Comment (KurochkaR) [reply auto_bot/handlers/order/handlers.py:260]: ok i'll refactor

File: auto_bot/handlers/order/handlers.py (hunk 3 above)
Comment (KurochkaR) [reply auto_bot/handlers/order/handlers.py:204]: i fixed omw

File: app/models.py (hunk 1 above)
Comment (SpaceIgor) [reply app/models.py:395]: for partner - he won’t know which fleet this rating belongs to

Comment (KurochkaR) [reply app/models.py:397]: it's not rating it's driver model

File: auto/tasks.py (hunk 6 above)
Comment (SpaceIgor) [reply auto/tasks.py:47]: how to pass partner new func

File: auto_bot/handlers/order/static_text.py (hunk 9 above)
Comment (SpaceIgor) [reply auto_bot/handlers/order/static_text.py:180]: use one var and create two different

File: selenium_ninja/synchronizer.py (hunk 10 above)
Comment (SpaceIgor) [reply selenium_ninja/synchronizer.py:58]: down u created driver without partner

File: auto/tasks.py (hunk 6 above)
Comment (KurochkaR) [reply auto/tasks.py:47]: if partner give us credentials for (uber or bolt or uklon or gps) than tasks that using this credentials will use for this partner only classes that partner use (ex. uber, bolt or only uklon)

File: auto_bot/handlers/order/static_text.py (hunk 9 above)
Comment (KurochkaR) [reply auto_bot/handlers/order/static_text.py:180]: method client_order_info use order object as requiered arg i add time_update (not requiered) for checking when user not find driver and reuse his order on different time i use both args in one case

File: app/models.py (hunk 1 above)
Comment (SpaceIgor) [reply app/models.py:397]: okay

Comment (SpaceIgor) [reply app/models.py:397]: ok sr

File: selenium_ninja/synchronizer.py (hunk 10 above)
Comment (KurochkaR) [reply selenium_ninja/synchronizer.py:58]: driver = Driver.objects.create(name=kwargs['name'],
                                           second_name=kwargs['second_name'],
                                           phone_number=kwargs['phone_number'],
                                           email=kwargs['email'],
                                           vehicle=self.get_or_create_vehicle(**kwargs),
                                           role=Role.DRIVER,
                                           _partner=Partner.get_partner(self.partner_id))_

File: selenium_ninja/synchronizer.py (hunk 4 above)
Comment (SpaceIgor) [reply selenium_ninja/synchronizer.py:108]: ok

File: auto/tasks.py (hunk 6 above)
Comment (SpaceIgor) [reply auto/tasks.py:47]: how management func for partners (for ex if the partner does not need efficiency or others)

File: auto_bot/handlers/order/static_text.py (hunk 9 above)
Comment (SpaceIgor) [reply auto_bot/handlers/order/static_text.py:180]: message = "Замовлення оновлено\n" 181
u need create global var
order_update = "Замовлення оновлено\n"

first if:
    mess = f{order_update}
                 f"Нова сума замовлення: {order.sum} грн\n"
second:
     mess = order_update

File: auto/tasks.py (hunk 6 above)
Comment (KurochkaR) [reply auto/tasks.py:47]: we give him all our functionality he can use it or not

Comment (SpaceIgor) [reply auto/tasks.py:47]: we must have functions for management(because it's all func cost money)

Comment (KurochkaR) [reply auto/tasks.py:47]: I haven't that in my issue

File: selenium_ninja/synchronizer.py (hunk 10 above)
Comment (SpaceIgor) [reply selenium_ninja/synchronizer.py:58]: ok

File: auto_bot/handlers/order/static_text.py (hunk 9 above)
Comment (KurochkaR) [reply auto_bot/handlers/order/static_text.py:180]: As u want

Comment (SpaceIgor) [reply auto_bot/handlers/order/static_text.py:180]: yeap

File: auto/tasks.py (hunk 6 above)
Comment (SpaceIgor) [reply auto/tasks.py:47]: why u have global dict with fleets?
u can have relation to partner and when use loops for run tasks

Comment (KurochkaR) [reply auto/tasks.py:47]: ok implement this in next pr

**Review by SpaceIgor — APPROVED** [review]

**Pull request closed without merging**